hphp/runtime/vm/jit/simplify.cpp

```diff
@@ -742,6 +742,27 @@ SSATmp* simplifyMulDbl(State& env, const IRInstruction* inst) {
   return constImpl(env, inst->src(0), inst->src(1), std::multiplies<double>());
 }
 
+SSATmp* simplifyMulIntOWithConstant(State& env, SSATmp* s, int64_t c)
+{
+  // X * 0 --> 0
+  if (c == 0)
+    return cns(env, 0);
+
+  // X * 1 --> X
+  if (c == 1)
+    return s;
+
+  // X * (-1) --> (0 - X)
+  if (c == -1)
+    return gen(env, SubInt, cns(env, 0), s);
```

Comment:
Turns out that SubIntO on AArch64 doesn't handle that right either... so I'll need to add additional test cases and fix the SubIntO while doing this one.